src/components/ZChart/ZChart.vue

```diff
@@ -173,50 +173,17 @@ export default Vue.extend({
     this.themeColors = this.getThemeColors(baseColors)
   },
   watch: {
-    chartType() {
-      this.initChart()
-    },
-    labels() {
-      this.updateChart()
-    },
-    dataSets() {
-      this.updateChart()
-    },
-    yMarkers() {
-      this.updateChart()
+    chartData: {
+      handler: function () {
+        this.updateChart()
+      }
+      // deep: true
```

Comment:
Did you intend to leave this commented out? I'd reckon we'd need `deep` to be true here, since `chartData` is a deeply nested object.

Reply:
@vaibhav-deepsource Nope, since it's a computed property, it will be updated correctly every time some property within it changes. We don't want to watch properties inside the properties, e.g. a data point inside the array because updating it that way wouldn't be reactive normally.